File: python/snippets/getting_started_generate_clickhouse.py
from flyql import parse
from flyql.generators.clickhouse.generator import to_sql_where
from flyql.generators.clickhouse.column import Column

result = parse("status >= 400 and host = prod*")

columns = {
    "status": Column("status", False, "UInt32"),
    "host": Column("host", False, "String"),
}

sql = to_sql_where(result.root, columns)
print(sql)

File: javascript/snippets/getting-started-generate-clickhouse.mjs
import { parse } from 'flyql'
import { generateWhere, newColumn } from 'flyql/generators/clickhouse'

const result = parse("status >= 400 and host like 'prod%'")

const columns = {
    status: newColumn("status", false, "UInt32", null),
    host: newColumn("host", false, "String", null),
}

const sql = generateWhere(result.root, columns)
console.log(sql)
